### PR TITLE
Update Test Cases to Align with New Error Handling for Missing Project/Dataset

### PR DESCRIPTION
**Pull request by yogendrasinghx**
Following the recent changes introduced in the error handling process, where missing projects or datasets now trigger an error message instead of an exception, I've noticed that this update has affected the behavior of some existing test cases. Specifically, `test_dataset_004` and `test_project_004` were failing because they expected an exception to confirm the deletion of a missing dataset or project.

To address this, I have adjusted these test cases to align with the new error handling strategy. This update ensures that the tests now check for the correct error messages rather than expecting an exception.Additionally, I've refactored the naming convention of the `getLogger` method to align with Python conventions, renaming it to `get_logger`.

Please review the changes and let me know if further modifications are needed. Your feedback is appreciated!

I have read the CLA Document and I sign the CLA

Resolves: UHUB-87

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced error handling and logging in dataset and project management modules.

### 📊 Key Changes
- 🛠 Added more robust error handling and logging for dataset and project fetch operations.
- 🧪 Updated test cases to use a more streamlined logging setup and improved checks.
- 🚀 Improved checks in test cases to accurately reflect the presence of data.

### 🎯 Purpose & Impact
- **Error Handling:** The introduction of try-except blocks around data retrieval operations ensures that errors are caught and logged, making the system more reliable and easier to debug. 🛡️
- **Better Logging:** Changing `getLogger` to `get_logger` across test files standardizes logging and simplifies the process of tracking actions and errors, contributing to more maintainable code. 📝
- **Data Checking Enhancements:** Updates in test cases to verify the existence and non-emptiness of data lead to more accurate tests, ensuring that functionalities are rigorously validated before deployment. ✅
- **User Impact:** These changes are primarily under-the-hood improvements aimed at developers and maintainers of the Ultralytics Hub SDK, leading to a more stable and dependable platform for end users building and managing datasets and projects. 🌐

### CODE REVIEW

**Review by sergiuwaxmann — APPROVED** [review]

**Comment by glenn-jocher** [comment]
@yogendrasinghx nice camelCase fixes. 

One simplification you can make is to skip the if else statement. Can you update the PR with this change?

```diff
-            if dataset_data is not None and len(dataset_data) > 0:
-                return True
-            else:
-                return False

+            return dataset_data is not None and len(dataset_data)
```

File: tests/features/dataset.py [hunk 1]
```diff
@@ -46,10 +46,14 @@ def is_dataset_exists(self, dataset_id):
             bool: True if the dataset exists, False otherwise.
         """
         try:
-            self.get_dataset_by_id(dataset_id)
-            return True
+            dataset = self.get_dataset_by_id(dataset_id)
+            dataset_data = dataset.data
+            if dataset_data is not None and len(dataset_data) > 0:
```
Comment (glenn-jocher) [inline tests/features/dataset.py:51]: remove if else and return directly, both here and in project.py.

Comment (yogendrasinghx) [reply tests/features/dataset.py:51]: Sure, I can do that. Let me update the PR with this change.

Comment (UltralyticsAssistant) [reply tests/features/dataset.py]: Great!

**Pull request merged**